### PR TITLE
Seirmaster

### PR DESCRIPTION
**Pull request by ovese**
Modified the seir model to only nonlinear model.

### CODE REVIEW

**Comment by ovese** [comment]
The modified SEIR model has passed all checks and dynamic response is good also

**Comment by ovese** [comment]
please find commit for the filter. it has passed the checks. IS this ok for patch 1 or I need to add more content to the code

File: app/gui/node-editor.cpp [hunk 1]
```diff
@@ -3077,21 +3088,6 @@ editor::show_window() noexcept
                       3,
                       "Fail to initialize example_qss_izhikevich<1>: %s\n",
                       status_string(ret));
-            if (ImGui::MenuItem("Insert example QSS1 seir_lineaire"))
```
Comment (quesnel) [inline app/gui/node-editor.cpp:3080]: I prefer a second pull request specific for this update of the SEIR model.

File: app/gui/node-editor.cpp [hunk 2]
```diff
@@ -3126,6 +3122,13 @@ editor::show_window() noexcept
                       3,
                       "Fail to initialize example_qss_izhikevich<2>: %s\n",
                       status_string(ret));
+            if (ImGui::MenuItem("Insert example QSS2 seir_nonlinear"))
```
Comment (quesnel) [inline app/gui/node-editor.cpp:3125]: I prefer a second pull request specific for this update of the SEIR model.

File: lib/include/irritator/core.hpp [hunk 3]
```diff
@@ -5218,6 +5219,23 @@ struct constant
         return { value };
     }
 };
+//starts the filter +++++++++++++++++++++++++++++++++
```
Comment (quesnel) [inline lib/include/irritator/core.hpp:5222]: This comment is unnecessary.

File: lib/include/irritator/core.hpp [hunk 4]
```diff
@@ -5218,6 +5219,23 @@ struct constant
...
+struct filter
+{
+    port x[1];
+    port y[1];
+    time sigma;
+    i64 number;
```
Comment (quesnel) [inline lib/include/irritator/core.hpp:5228]: You can remove this useless attribute.

File: lib/include/irritator/core.hpp [hunk 5]
```diff
@@ -5218,6 +5219,23 @@ struct constant
...
+
+    status initialize() noexcept
+    {
+        sigma=time_domain<time>::infinity;
+        number ={0}; 
+        return status::success;
+    }
+
```
Comment (quesnel) [inline lib/include/irritator/core.hpp:5236]: You can remove this empty line.

File: lib/include/irritator/examples.hpp [hunk 6]
```diff
@@ -304,98 +304,34 @@ example_qss_negative_lif(simulation& sim, F f) noexcept
 
 template<int QssLevel, typename F>
 status
-example_qss_seir_lineaire(simulation& sim, F f) noexcept
```
Comment (quesnel) [inline lib/include/irritator/examples.hpp:307]: I prefer a second pull request specific for this update of the SEIR model.

**Review by quesnel — COMMENTED** [review]
Please, can you provide two pull request ? First for the SEIR model update, second for the filter model.

**Comment by ovese** [comment]
closing this pull request due to mixed patch issues

**Pull request closed without merging**